simpleclient/src/main/java/io/prometheus/client/DoubleAdder.java

```diff
@@ -118,6 +142,38 @@ public void reset() {
         internalReset(0L);
     }
 
+    public void set(double x) {
+        for (;;) {
+            Cell[] as;
+            if ((as = cells) != null) { // have cells
+                if (busy == 0 && casBusy()) {
+                    try {
+                        if (cells == as) { // recheck under lock
+                            // allocate new cells instead of updating existing cells
+                            int n = as.length;
+                            Cell[] rs = new Cell[n];
+                            for (int i = 0; i < n; ++i) {
+                                Cell a = as[i];
+                                if (a != null) // avoid unused cells
+                                    rs[i] = a.value == 0L ? a : new Cell(0L); // reuse or initialize cell
+                            }
+                            // update cells and base (not atomic)
```

Comment:
Is this still true?

Reply:
It depends on how we look at it. 😉 
It's not written atomically, but it's read "atomically". 

On concurrent `get` and `set` and `inc / dec`:
* it is possible that `set` writes `cells`, then `inc / dec` writes  `base` (no contention), then `set` writes `base`
* it is possible that `sum` reads intermediate state, but it detects this and retries, so it won't use invalid state

Reply:
The comment as written hints that there's a race here, but there isn't.